google/cloud/aiplatform/utils/__init__.py

```diff
@@ -325,6 +325,34 @@ def extract_bucket_and_prefix_from_gcs_path(gcs_path: str) -> Tuple[str, Optiona
     return (gcs_bucket, gcs_blob_prefix)
 
 
+def extract_project_and_location_from_parent(
+    parent: str,
+) -> Dict[Optional[str], Optional[str]]:
```

Comment:
`Optional[str]` is a synonym for `Union[str, None]`, which would imply that the return value could have `None` in the key or value.  The return type should simply be `Dict[str, str]` here.

Reply:
The function returns either a `Dict[str, str]` or an empty dict `{}`. Does having `Dict[str, str]` also capture `{}`. But sure, I'll add the change

Reply:
That is correct, the typing annotation `Dict[str, str]` includes the empty dict.

The confusion here is that the word `Optional` does not mean what it means in English: https://docs.python.org/3/library/typing.html#typing.Optional, specifically "Note that this is not the same concept as an optional argument..."

Reply:
Gotcha, thanks Yuhan